shimming-toolbox/shimmingtoolbox/cli/b0shim.py

```diff
@@ -227,9 +230,18 @@ def dynamic(fname_fmap, fname_anat, fname_mask_anat, method, opt_criteria, slice
     with open(fname_anat_json) as json_file:
         json_anat_data = json.load(json_file)
 
-    # Get the EPI echo time if optimization criteria is grad
-    if opt_criteria == 'grad':
+    # Get the EPI echo time and set signal recovery optimizer criteria if w signal loss is set
+    if w_signal_loss or w_signal_loss_xy:
+        if opt_criteria != 'mse':
```

Comment:
rmse as well?

Reply:
fixed in [2ae6977](https://github.com/shimming-toolbox/shimming-toolbox/pull/573/commits/2ae6977d3f96b0472bc2adb6546a6d23edca160d)